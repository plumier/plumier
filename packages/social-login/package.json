{
  "name": "@plumier/social-login",
  "version": "1.0.0-beta.10",
  "description": "OAuth module for Plumier",
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "scripts": {
    "compile": "tsc -p tsconfig.build.json"
  },
  "author": "Ketut Sandiarsa",
  "license": "MIT",
  "dependencies": {
    "@plumier/core": "1.0.0-beta.10",
    "@types/axios": "0.14.0",
    "@types/debug": "^4.1.5",
    "axios": "0.19.0",
    "debug": "^4.1.1"
  },
  "bugs": {
    "url": "https://github.com/plumier/plumier/issues"
  },
  "repository": {
    "url": "https://github.com/plumier/plumier"
  },
  "homepage": "https://plumierjs.com",
  "publishConfig": {
    "access": "public"
  },
  "devDependencies": {
    "@types/dotenv": "^8.2.0",
    "dotenv": "^8.2.0"
  }
}
