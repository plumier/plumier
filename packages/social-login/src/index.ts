import { ActionResult, middleware, response } from "@plumier/core"

import { content } from "./callback-view"
import { DialogProvider, SocialAuthProvider, OAuthCallbackMiddleware, OAuthDialogEndPointMiddleware } from "./middleware"

export function oAuthCallback(option: SocialAuthProvider) {
    return middleware.use(new OAuthCallbackMiddleware(option))
}

export function oAuthDialogEndPoint(option:DialogProvider){
    return middleware.use(new OAuthDialogEndPointMiddleware(option))
}

export { GoogleProvider, GoogleLoginStatus, GoogleProfile, GoogleDialogProvider } from "./provider/google"
export { FacebookProvider, FacebookLoginStatus, FacebookProfile, FacebookDialogProvider } from "./provider/facebook"
export { GitHubProvider, GitHubLoginStatus, GitHubProfile, GitHubDialogProvider } from "./provider/github"
export { GitLabProvider, GitLabLoginStatus, GitLabProfile, GitLabDialogProvider } from "./provider/gitlab"
export { OAuthCallbackMiddleware, OAuthDialogEndPointMiddleware, SocialLoginStatus, SocialAuthProvider, DialogProvider } from "./middleware"

declare module "@plumier/core" {
    namespace response {
        function callbackView(message: any): ActionResult;
    }
}

response.callbackView = (message: any) => {
    return new ActionResult(content(message))
        .setHeader("Content-Type", "text/html")
}