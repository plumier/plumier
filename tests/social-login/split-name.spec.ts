import { splitName } from "@plumier/social-login"

describe("splitName", () => {
    it("Should split name with 4 words", () => {
        expect(splitName("I Ketut Sandiarsa Rugita")).toMatchSnapshot()
    })
    it("Should split name with 3 words", () => {
        expect(splitName("Ketut Sandiarsa Rugita")).toMatchSnapshot()
    })
    it("Should split name with 2 words", () => {
        expect(splitName("Ketut Sandiarsa")).toMatchSnapshot()
    })
    it("Should split name with 1 words", () => {
        expect(splitName("Ketut")).toMatchSnapshot()
    })
    it("Should not error when provided undefined", () => {
        expect(splitName(undefined)).toMatchSnapshot()
    })
})
