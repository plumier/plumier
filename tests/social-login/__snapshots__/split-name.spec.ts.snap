// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`splitName Should not error when provided undefined 1`] = `
Object {
  "firstName": "",
  "lastName": "",
}
`;

exports[`splitName Should split name with 1 words 1`] = `
Object {
  "firstName": "Ketut",
  "lastName": "",
}
`;

exports[`splitName Should split name with 2 words 1`] = `
Object {
  "firstName": "Ketut",
  "lastName": "Sandiarsa",
}
`;

exports[`splitName Should split name with 3 words 1`] = `
Object {
  "firstName": "Ketut Sandiarsa",
  "lastName": "Rugita",
}
`;

exports[`splitName Should split name with 4 words 1`] = `
Object {
  "firstName": "I Ketut Sandiarsa",
  "lastName": "Rugita",
}
`;
